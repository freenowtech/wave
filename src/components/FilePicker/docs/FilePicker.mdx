---
name: FilePicker
menu: Components
route: /components/file-picker
---

import { Playground } from 'docz';
import { StyledSystemLinks } from '../../../docs/StyledSystemLinks';
import { FilePicker, Button, Box } from '../'
import { ControlledErrorFilePicker } from './ControlledErrorFilePicker'
import { FilePickerPropsTable } from './FilePickerPropsTable';


# FilePicker

This component itâ€™s a trigger opening the native file uploader.
This component uses default Web input type file check [Mozilla Input Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file)
if you need to know more about the supported attributes and values

## Properties

<FilePickerPropsTable />
<br />
<StyledSystemLinks component="FilePicker" supportedProps={['margin']} />

## Appearance
- **File title**: This provides a clear description of what the user should upload. If long, takes as many lines as needed extending the height of the component.
- **Button label**: Describes the main action, it works better when short
- It is recommended to place this component together with a helper text to provide additional help for what's expected. Helper text should only appear if the user open the file explorer and doesn't select anything.

## Placement

- They are mainly used in forms but can also appear in other places as a stand alone component, like item or user registration screens.
- It should be placed with a minimum of 8px _("Spaces" value 1 for our default spaces values)_ margin on each side
- It should be placed vertically aligned

## Behaviour

- The user may select 1 file at a time.
- By default, any file format is accepted, but you can add parameters to validate a specific file format.
- The action of clicking Browse file will trigger a browser-specific upload window.
- Once the user selects the file, the browser-specific select window closes and the files will appear below the File title label.

### Error
Below you have an example of how you can properly show an error using the **HelperText**. You can try to submit any file to trigger the error
and clean the input by canceling the selection to remove the error state.

<ControlledErrorFilePicker />

## Playground
The example below will demostrate how to create compositions of `FilePicker`

<Playground>
  <Box maxWidth="100%" width="30rem" mx="auto" p={2}>
    <FilePicker
      accept="image/png, image/jpeg"
      buttonText="Browse files"
      label="A picture of you"
      name="avatar"
      onChange={(e) => console.log('onChange', e)}
      onFileChange={(file, e) => console.log('onFileChange', file, e)}
    />
    <FilePicker
      accept="image/png, image/jpeg"
      buttonText="Browse files"
      label="A logo you like"
      name="logo"
      onChange={(e) => console.log('onChange', e)}
      onFileChange={(file, e) => console.log('onFileChange', file, e)}
      mt={1}
    />
    <FilePicker
      accept="image/png, image/jpeg"
      buttonText="Browse files"
      label="This provides a clear description of what the user should upload. Can be multiline"
      name="logo"
      onChange={(e) => console.log('onChange', e)}
      onFileChange={(file, e) => console.log('onFileChange', file, e)}
      mt={1}
    />
  </Box>
</Playground>

In the real world you may need to deal with the `FilePicker` within a form that will handle the submission of the file. Below you will find a snippet
that can help as example.

**Tip:** The `FilePicker` exposes an additional callback of `onFileChange` which signature is `(file, event)`. The [File](https://developer.mozilla.org/es/docs/Web/API/File)
and the standard input event. This is a convenient to use `file` instead of `e.target.elements.avatar.files[0]` check the console while using the example below.

<Playground>
  <Box maxWidth="100%" width="30rem" mx="auto">
    <form onSubmit={(e) => {
      alert(`
        About to submit ${e.target.elements.avatar.value}.
        Check console to see the value provided by the Web File API.
        more at https://developer.mozilla.org/es/docs/Web/API/File
      `)
      console.log(e.target.elements.avatar.files[0])
      e.preventDefault()
    }}>
      <Box mb={2}>
        <FilePicker
          accept="image/png, image/jpeg"
          buttonText="Browse files"
          label="A picture of you"
          name="avatar"
          onChange={(e) => console.log('onChange', e)}
          onFileChange={(file, e) => console.log('onFileChange', file, e)}
        />
      </Box>
      <Button type="submit">Submit</Button>
    </form>
  </Box>
</Playground>
