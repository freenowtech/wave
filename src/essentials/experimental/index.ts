export * from './theme';
export { createGlobalStyle, ColorPalette } from './Colors';
export { darkTheme } from './globalStyles';
export { getSemanticValue, getBareValue, getBareHslValue, getSemanticHslValue } from './cssVariables';
