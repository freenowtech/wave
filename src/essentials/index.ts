export { Elevation } from './Elevation/Elevation';
export { Spaces } from './Spaces/Spaces';
export { Breakpoints, MediaQueries } from './Breakpoints/Breakpoints';
export type { ReadCssColorVariable, ReadCssVariable } from './Colors/types';
export { GlobalStyle as ClassicColors } from './Colors/Colors';
export { GlobalStyle as ModernColors } from './Colors/ModernColors';
