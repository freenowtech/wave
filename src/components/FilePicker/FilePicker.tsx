import React, { ComponentPropsWithoutRef, FC, useRef, useState } from 'react';
import styled, { css } from 'styled-components';
import { MarginProps } from 'styled-system';
import { Colors, MediaQueries } from '../../essentials';
import { CheckCircleOutlineIcon, ShareIcon } from '../../icons';
import { get } from '../../utils/themeGet';
import { Box, BoxProps } from '../Box/Box';
import { Button } from '../Button/Button';
import { Text } from '../Text/Text';
import { shrinkFileName } from './utils/format';

interface FilePickerProps extends MarginProps, ComponentPropsWithoutRef<'input'> {
    /**
     * Sets the name property of input element in the DOM.
     * https://developer.mozilla.org/en-US/docs/Web/API/Element/name
     */
    name: string;
    /**
     * One or more unique file type identifiers describing
     * files types to allow.
     * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file
     */
    accept?: string;
    /**
     * What source to use for capturing image or video data.
     * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file
     */
    capture?: string;
    /**
     * Text to provide a clear description of what the user should upload
     */
    label: string;
    /**
     * Text to display within component button, it describes the main action,
     * it works better when short
     */
    buttonText: string;
    /**
     * Weather or not the component should render a error state
     */
    error?: boolean;
    /**
     * Callback to be used when the input receives a new file
     */
    onFileChange?: (file: File, event: React.ChangeEvent<HTMLInputElement>) => void;
}

const InputButton = styled(Button)`
    font-size: ${get('fontSizes.1')};
    line-height: 1;
    font-weight: 400;
    height: auto;
    min-width: auto;
    padding: 0.5rem 0.75rem;
    white-space: nowrap;
`;

const ICON_FILE_FEEDBACK_COLOR = Colors.POSITIVE_GREEN_900;

interface OutlinerProps extends BoxProps {
    disabled: boolean;
    error: boolean;
    hasValidFile: boolean;
    onClick: (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
}

// NOTE: we want to affect the color of only one icon SVG and not the ICON_FILE_FEEDBACK_COLOR
const Outliner = styled(Box)<OutlinerProps>`
    border: 0.0625rem solid ${Colors.AUTHENTIC_BLUE_200};
    box-sizing: border-box;
    cursor: pointer;
    position: relative;
    transition: background-color ease 200ms, border-color ease 200ms;

    ${Text} {
        cursor: pointer;
    }

    ${({ disabled }) =>
        disabled &&
        css`
            opacity: 0.5;

            &,
            ${InputButton}, ${Text} {
                cursor: not-allowed;
            }
        `}

    ${({ error }) =>
        error &&
        css`
            box-shadow: inset 0 0 0 0.0625rem ${Colors.NEGATIVE_ORANGE_900};
            border-color: ${Colors.NEGATIVE_ORANGE_900};
        `}

    ${({ hasValidFile }) =>
        hasValidFile &&
        css`
                ${MediaQueries.medium} {
                    &:hover {
                        background-color: ${Colors.ACTION_BLUE_50};
                        border-color: ${Colors.ACTION_BLUE_50};

                        svg:not([color='${Colors.POSITIVE_GREEN_900}']) path {
                            fill: ${Colors.ACTION_BLUE_900};
                        }
                    }
                }
            `}
`;

const Input = styled.input`
    height: 100%;
    left: 0;
    opacity: 0;
    position: absolute;
    top: 0;
    width: 100%;
    outline: none;

    &:focus + ${Outliner} {
        border-color: ${Colors.ACTION_BLUE_900};
        box-shadow: inset 0 0 0 0.0625rem ${Colors.ACTION_BLUE_900};
    }
`;

const FilePicker: FC<FilePickerProps> = ({
    name,
    accept = '*',
    capture,
    label,
    buttonText,
    error = false,
    onFileChange = () => undefined,
    onChange = () => undefined,
    disabled = false,
    ...nonInputProps
}: FilePickerProps) => {
    const inputEl = useRef<HTMLInputElement>(null);
    const [file, setFile] = useState<File | null>();
    const onInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
        const eventFile = e.target.files?.[0];

        onChange(e);
        onFileChange(eventFile, e);
        setFile(eventFile);
    };
    const onClickHandler = (e: React.MouseEvent<HTMLButtonElement>) => {
        // Avoid label trigger file selection twice
        e.preventDefault();
        // Avoid button trigger file selection twice
        e.stopPropagation();
        // Allow other spaces of the component trigger file selection
        inputEl.current.click();
    };
    const validFileSelected = file && !error;

    return (
        <Box position="relative" {...nonInputProps}>
            <Input
                accept={accept}
                capture={capture}
                disabled={disabled}
                id={name}
                multiple={false}
                name={name}
                onChange={onInputChange}
                ref={inputEl}
                type="file"
            />
            <Outliner
                minHeight="5.125rem"
                alignItems="stretch"
                data-testid="filepicker-outliner"
                display="flex"
                disabled={disabled}
                hasValidFile={validFileSelected}
                error={error}
                justifyContent="space-between"
                onClick={onClickHandler}
                px={2}
                py={validFileSelected ? '1.25rem' : 3}
            >
                <Box display="flex" flexDirection="column" justifyContent="center" pr={2}>
                    <Text as="label" htmlFor={name} fontSize={1}>
                        {label}
                    </Text>
                    <Text weak fontSize={1}>
                        {file && shrinkFileName(file)}
                    </Text>
                </Box>
                <Box display={{ _: 'none', medium: 'flex' }} alignItems="center" justifyContent="top">
                    {validFileSelected ? (
                        <CheckCircleOutlineIcon color={ICON_FILE_FEEDBACK_COLOR} />
                    ) : (
                        <InputButton variant="secondary" type="button" tabIndex={-1} onClick={onClickHandler}>
                            {buttonText}
                        </InputButton>
                    )}
                </Box>
                <Box display={{ _: 'flex', medium: 'none' }} alignItems="top">
                    {validFileSelected ? <CheckCircleOutlineIcon color={ICON_FILE_FEEDBACK_COLOR} /> : <ShareIcon />}
                </Box>
            </Outliner>
        </Box>
    );
};

export { FilePicker, FilePickerProps };
