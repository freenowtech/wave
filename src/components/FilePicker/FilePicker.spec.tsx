import React from 'react';
import { render as renderRtl, screen, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { FilePicker, FilePickerProps } from './FilePicker';
import { Colors } from '../../essentials';

const baseProps = {
    label: 'A picture of you',
    buttonText: 'Browse files',
    name: 'avatar',
    accept: 'image/png, image/jpeg'
};

/**
 * * Encourage the test of the component within the integration
 * of itself in the common target scenario that could be used.
 *
 * @param filePickerProps the filePicker accepted props
 * @param onFormSubmit a callback to handle a simulation of a form submit
 */
const render = (filePickerProps: FilePickerProps, onFormSubmit = jest.fn()) => {
    const Form = (props: FilePickerProps) => (
        <form onSubmit={onFormSubmit}>
            <FilePicker {...props} data-testid="file-picker" />
            <button type="submit">Submit</button>
        </form>
    );

    return renderRtl(<Form {...filePickerProps} />);
};

test('renders a composition with an input type file', () => {
    render({ ...baseProps, onFileChange: jest.fn() });

    expect(screen.getByTestId('file-picker')).toMatchSnapshot();
});

test('when a file gets submitted component removes the initial button', () => {
    const file = new File(['(⌐□_□)'], 'carnet.png', { type: 'image/png' });
    const onFileChange = jest.fn();
    const buttonText = 'select file';

    render({ ...baseProps, onFileChange, buttonText });

    expect(screen.queryByText(buttonText)).toBeInTheDocument();

    // https://github.com/kentcdodds/react-testing-library-examples/pull/1/files
    const inputEl = screen.getByLabelText(/picture of you/);
    Object.defineProperty(inputEl, 'files', { value: [file] });

    fireEvent.change(inputEl);

    expect(screen.queryByText(buttonText)).not.toBeInTheDocument();
    expect(screen.getByTestId('file-picker')).toMatchSnapshot();
});

test('expose a way to handle errors', () => {
    const file = new File(['(⌐□_□)'], 'carnet.png', { type: 'image/png' });
    const onFormSubmit = jest.fn();
    const onError = jest.fn();
    const onFileChange = jest.fn().mockImplementation((eventFile: File) => {
        try {
            if (eventFile.size > 0) {
                throw new Error('Oops! I need a file with no size');
            }
        } catch {
            onError(eventFile);
        }
    });

    render({ ...baseProps, onFileChange }, onFormSubmit);

    // https://github.com/kentcdodds/react-testing-library-examples/pull/1/files
    const inputEl = screen.getByLabelText(/picture of you/);
    Object.defineProperty(inputEl, 'files', { value: [file] });

    fireEvent.change(inputEl);

    expect(onError).toHaveBeenCalledTimes(1);
});

test('expose a way to get selected file within the host composition', () => {
    const file = new File(['(⌐□_□)'], 'carnet.png', { type: 'image/png' });
    let submittedFiles: FileList;
    const onFileChange = jest.fn();
    // emulate real environment and make sure FilePicker properly support forms
    const onFormSubmit = jest.fn().mockImplementation((event: React.FormEvent<HTMLFormElement>) => {
        event.preventDefault();
        submittedFiles = ((event.target as HTMLFormElement).elements.namedItem('avatar') as HTMLInputElement).files;
    });

    render({ ...baseProps, onFileChange }, onFormSubmit);

    // https://github.com/kentcdodds/react-testing-library-examples/pull/1/files
    const inputEl = screen.getByLabelText(/picture of you/);
    Object.defineProperty(inputEl, 'files', { value: [file] });

    fireEvent.change(inputEl);
    userEvent.click(screen.getByText(/submit/i));

    expect(onFileChange).toHaveBeenCalledTimes(1);
    // Assert over SyntheticEvent will lead to console errors
    expect(onFileChange).toHaveBeenCalledWith(file, expect.anything());
    expect(onFormSubmit).toHaveBeenCalledTimes(1);
    expect(submittedFiles[0]).toEqual(file);
});

test('expose a way to access the event using onChange input standard prop', () => {
    const file = new File(['(⌐□_□)'], 'carnet.png', { type: 'image/png' });
    const onFileChange = jest.fn();
    const onChange = jest.fn();
    renderRtl(<FilePicker {...baseProps} onFileChange={onFileChange} onChange={onChange} data-testid="file-picker" />);

    const inputEl = screen.getByLabelText(/picture of you/);
    Object.defineProperty(inputEl, 'files', { value: [file] });

    fireEvent.change(inputEl);

    expect(onFileChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledTimes(1);
    // Instead we expect the SyntheticEvent to be passed
    expect(onChange.mock.calls[0]).not.toEqual(file);
});

test('accepts an error prop that will change the component appareance', () => {
    const file = new File(['(⌐□_□)'], 'carnet.png', { type: 'image/png' });
    const onFileChange = jest.fn();

    const { rerender } = renderRtl(<FilePicker {...baseProps} onFileChange={onFileChange} />);

    const inputEl = screen.getByLabelText(/picture of you/);
    const outliner = screen.getByTestId('filepicker-outliner');
    Object.defineProperty(inputEl, 'files', { value: [file] });

    fireEvent.change(inputEl);

    rerender(<FilePicker {...baseProps} onFileChange={onFileChange} error />);

    expect(screen.queryByTitle(/check circle/i)).not.toBeInTheDocument();
    expect(outliner).toHaveStyle({
        'border-color': Colors.NEGATIVE_ORANGE_900
    });
});

test('accepts an disabled prop that will change the component appareance', () => {
    const onFileChange = jest.fn();

    const { container } = renderRtl(<FilePicker {...baseProps} onFileChange={onFileChange} disabled />);

    const inputEl = screen.getByLabelText(/picture of you/);
    const outliner = screen.getByTestId('filepicker-outliner');

    expect(inputEl).toBeDisabled();
    expect(outliner).toHaveStyle({
        opacity: '0.5'
    });
    expect(container).toMatchSnapshot();
});
