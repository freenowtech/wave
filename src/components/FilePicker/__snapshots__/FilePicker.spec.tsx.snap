// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`accepts an disabled prop that will change the component appareance 1`] = `
.c0 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 1rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
  -webkit-box-pack: top;
  -webkit-justify-content: top;
  -ms-flex-pack: top;
  justify-content: top;
}

.c11 {
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
  font-size: 0.875rem;
}

.c7 {
  color: #637689;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
  font-size: 0.875rem;
}

.c3 {
  min-height: 5.125rem;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  border: 0.0625rem solid #C6CDD4;
  box-sizing: border-box;
  cursor: pointer;
  position: relative;
  -webkit-transition: background-color ease 200ms,border-color ease 200ms;
  transition: background-color ease 200ms,border-color ease 200ms;
  opacity: 0.5;
}

.c3 .c5 {
  cursor: pointer;
}

.c3,
.c3 .c9,
.c3 .c5 {
  cursor: not-allowed;
}

.c1 {
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  width: 100%;
  outline: none;
}

.c1:focus + .c2 {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c10 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #001E3E;
  fill: #001E3E;
  background: #FFFFFF;
  border-color: #C6CDD4;
  font-size: 0.875rem;
  line-height: 1;
  font-weight: 400;
  height: auto;
  min-width: auto;
  padding: 0.5rem 0.75rem;
  white-space: nowrap;
}

.c10:disabled {
  cursor: not-allowed;
}

.c10 svg path {
  fill: inherit;
}

.c10:hover {
  color: #001E3E;
  fill: #001E3E;
  background: #F1F2F4;
  border-color: #C6CDD4;
}

.c10:disabled {
  color: #C6CDD4;
  fill: #C6CDD4;
  background: #FFFFFF;
  border-color: #C6CDD4;
}

@media screen and (min-width:48rem) {
  .c8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:48rem) {
  .c11 {
    display: none;
  }
}

<div>
  <div
    class="c0"
  >
    <input
      accept="image/png, image/jpeg"
      class="c1"
      disabled=""
      id="avatar"
      name="avatar"
      type="file"
    />
    <div
      class="c2 c3"
      data-testid="filepicker-outliner"
      disabled=""
      display="flex"
    >
      <div
        class="c4"
        display="flex"
      >
        <label
          class="c5 c6"
          font-size="1"
          for="avatar"
        >
          A picture of you
        </label>
        <span
          class="c5 c7"
          font-size="1"
        />
      </div>
      <div
        class="c8"
        display="[object Object]"
      >
        <button
          class="sc-AxjAm sc-AxirZ c9 c10"
          tabindex="-1"
          type="button"
        >
          Browse files
        </button>
      </div>
      <div
        class="c11"
        display="[object Object]"
      >
        <svg
          color="#001E3E"
          height="24"
          viewBox="0 0 24 24"
          width="24"
        >
          <path
            d="M19 19v2H5v-2h14zM12 3l6 5.333v2.675l-5-4.444V17h-2V6.564l-5 4.445V8.334L12 3z"
            fill="currentColor"
            fill-rule="nonzero"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders a composition with an input type file 1`] = `
.c0 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 1rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
  -webkit-box-pack: top;
  -webkit-justify-content: top;
  -ms-flex-pack: top;
  justify-content: top;
}

.c10 {
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
  font-size: 0.875rem;
}

.c7 {
  color: #637689;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
  font-size: 0.875rem;
}

.c3 {
  min-height: 5.125rem;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  border: 0.0625rem solid #C6CDD4;
  box-sizing: border-box;
  cursor: pointer;
  position: relative;
  -webkit-transition: background-color ease 200ms,border-color ease 200ms;
  transition: background-color ease 200ms,border-color ease 200ms;
}

.c3 .c5 {
  cursor: pointer;
}

.c1 {
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  width: 100%;
  outline: none;
}

.c1:focus + .c2 {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

.c9 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: transparent;
  border-radius: 0.25rem;
  border: 0.0625rem solid transparent;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: "Open Sans",sans-serif;
  font-weight: 600;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1rem;
  height: 3rem;
  min-width: 6.25rem;
  padding: 0 1rem;
  -webkit-transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  transition: background ease 200ms,border-color ease 200ms,color ease 200ms,fill ease 200ms;
  color: #001E3E;
  fill: #001E3E;
  background: #FFFFFF;
  border-color: #C6CDD4;
  font-size: 0.875rem;
  line-height: 1;
  font-weight: 400;
  height: auto;
  min-width: auto;
  padding: 0.5rem 0.75rem;
  white-space: nowrap;
}

.c9:disabled {
  cursor: not-allowed;
}

.c9 svg path {
  fill: inherit;
}

.c9:hover {
  color: #001E3E;
  fill: #001E3E;
  background: #F1F2F4;
  border-color: #C6CDD4;
}

.c9:disabled {
  color: #C6CDD4;
  fill: #C6CDD4;
  background: #FFFFFF;
  border-color: #C6CDD4;
}

@media screen and (min-width:48rem) {
  .c8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:48rem) {
  .c10 {
    display: none;
  }
}

<div
  class="c0"
  data-testid="file-picker"
>
  <input
    accept="image/png, image/jpeg"
    class="c1"
    id="avatar"
    name="avatar"
    type="file"
  />
  <div
    class="c2 c3"
    data-testid="filepicker-outliner"
    display="flex"
  >
    <div
      class="c4"
      display="flex"
    >
      <label
        class="c5 c6"
        font-size="1"
        for="avatar"
      >
        A picture of you
      </label>
      <span
        class="c5 c7"
        font-size="1"
      />
    </div>
    <div
      class="c8"
      display="[object Object]"
    >
      <button
        class="sc-AxjAm sc-AxirZ c9"
        tabindex="-1"
        type="button"
      >
        Browse files
      </button>
    </div>
    <div
      class="c10"
      display="[object Object]"
    >
      <svg
        color="#001E3E"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M19 19v2H5v-2h14zM12 3l6 5.333v2.675l-5-4.444V17h-2V6.564l-5 4.445V8.334L12 3z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`when a file gets submitted component removes the initial button 1`] = `
.c0 {
  position: relative;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 1rem;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: none;
  -webkit-box-pack: top;
  -webkit-justify-content: top;
  -ms-flex-pack: top;
  justify-content: top;
}

.c9 {
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  color: #001E3E;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
  font-size: 0.875rem;
}

.c7 {
  color: #637689;
  font-size: 1rem;
  font-family: "Open Sans",sans-serif;
  line-height: 1.4;
  margin: 0;
  font-size: 0.875rem;
}

.c10 .c5 {
  cursor: pointer;
}

.c3 {
  min-height: 5.125rem;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 1.25rem;
  padding-bottom: 1.25rem;
  border: 0.0625rem solid #C6CDD4;
  box-sizing: border-box;
  cursor: pointer;
  position: relative;
  -webkit-transition: background-color ease 200ms,border-color ease 200ms;
  transition: background-color ease 200ms,border-color ease 200ms;
}

.c3 .c5 {
  cursor: pointer;
}

.c1 {
  height: 100%;
  left: 0;
  opacity: 0;
  position: absolute;
  top: 0;
  width: 100%;
  outline: none;
}

.c1:focus + .c2 {
  border-color: #096BDB;
  box-shadow: inset 0 0 0 0.0625rem #096BDB;
}

@media screen and (min-width:48rem) {
  .c8 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (min-width:48rem) {
  .c9 {
    display: none;
  }
}

@media screen and (min-width:48rem) {
  .c3:hover {
    background-color: #F1F7FD;
    border-color: #F1F7FD;
  }

  .c3:hover svg:not([color='#069D4F']) path {
    fill: #096BDB;
  }
}

<div
  class="c0"
  data-testid="file-picker"
>
  <input
    accept="image/png, image/jpeg"
    class="c1"
    id="avatar"
    name="avatar"
    type="file"
  />
  <div
    class="c2 c3"
    data-testid="filepicker-outliner"
    display="flex"
  >
    <div
      class="c4"
      display="flex"
    >
      <label
        class="c5 c6"
        font-size="1"
        for="avatar"
      >
        A picture of you
      </label>
      <span
        class="c5 c7"
        font-size="1"
      >
        carnet.png
      </span>
    </div>
    <div
      class="c8"
      display="[object Object]"
    >
      <svg
        color="#069D4F"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm3.898 4.293l1.415 1.414-6.32 6.319-3.696-3.657 1.406-1.422 2.283 2.258 4.912-4.912z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </div>
    <div
      class="c9"
      display="[object Object]"
    >
      <svg
        color="#069D4F"
        height="24"
        viewBox="0 0 24 24"
        width="24"
      >
        <path
          d="M12 2c5.523 0 10 4.477 10 10s-4.477 10-10 10S2 17.523 2 12 6.477 2 12 2zm0 2a8 8 0 100 16 8 8 0 000-16zm3.898 4.293l1.415 1.414-6.32 6.319-3.696-3.657 1.406-1.422 2.283 2.258 4.912-4.912z"
          fill="currentColor"
          fill-rule="nonzero"
        />
      </svg>
    </div>
  </div>
</div>
`;
